modules/OTAR_CRISPR.py

```diff
@@ -0,0 +1,181 @@
+        """Processing the study dataframe"""
+
+        # Looping through the studies and generating evidence:
+        # Reading all data file and filter for significant hits:
+        hits = (
+            self.study_df[['studyId', 'dataFiles', 'dataFileType', 'filterColumn', 'threshold', 'projectId']]
+            .explode('dataFiles')
+            .assign(
+                dataFile=lambda df: df.apply(lambda x: f'{data_folder}/{x["projectId"]}/{x["dataFiles"]}', axis=1)
+            )
+            # TODO: parsing the data files should be file type dependent:
+            .apply(self.parse_MAGeCK_file, axis=1)
+        )
+
+        # Concatenate all hits into one single dataframe:
+        hits_df = (
+            pd.concat(hits.to_list())
+            .reset_index(drop=True)
+        )
+
+        # Merging:
+        self.merged_dataset = (
+            self.study_df
+            .assign(direction=lambda df: df.filterColumn.map(FILTER_COLUMN_MAP))
+            .merge(hits_df, on='studyId', how='inner')
+            .explode('diseaseFromSourceMappedId')
+            .assign(
+                datasourceId='ot_crispr',
+                datatypeId='ot_partner'
+            )
+            [[
+                'targetFromSourceId', 'diseaseFromSourceMappedId',
+                'projectId', 'studyId', 'studyOverview', 'contrast', 'crisprScreenLibrary',
+                'cellType', 'cellLineBackground', 'geneticBackground',
+                'direction', 'resourceScore',
+                'datasourceId', 'datatypeId'
+            ]]
+        )
+
+    @staticmethod
+    def parse_MAGeCK_file(row) -> pd.DataFrame:
+        """This function returns with a pandas dataframe with the datafile and with properly named columns"""
+
+        datafile = row['dataFile']
+        filterColumn = row['filterColumn']
+        threshold = float(row['threshold'])
+        studyId = row['studyId']
+
+        # Read data, filter and rename columns:
+        try:
+            mageck_df = (
+                pd.read_csv(datafile, sep='\t')
+                .rename(columns={filterColumn: 'resourceScore', 'id': 'targetFromSourceId'})
+                .loc[lambda df: df.resourceScore <= threshold]
+                [['targetFromSourceId', 'resourceScore']]
+                .assign(studyId=studyId)
+            )
+            logging.info(f'Number of genes reach threshold: {len(mageck_df)}')
+            return mageck_df
+        except FileNotFoundError:
+            logging.info(f'Study skipped as file was not found: {datafile}')
+
+    def get_evidence_data(self) -> pd.DataFrame:
+        """Returning the merged dataset"""
+        return self.merged_dataset
+
+    def write_evidence(self, output_file: str) -> None:
+        """Write the merged evidence to file"""
+
+        json_list = [json.dumps(row.dropna().to_dict()) for _, row in self.merged_dataset.iterrows()]
+        with gzip.open(output_file, 'wt') as f:
+            f.write('\n'.join(json_list))
```

Comment:
```suggestion
            f.write('\n'.join(json_list) + '\n')
```

To have that sweet final line break